rmf_scheduler_ros2_client/CMakeLists.txt

```diff
@@ -0,0 +1,73 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(rmf_scheduler_ros2_client)
+
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rmf_scheduler_msgs REQUIRED)
+
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic")
+endif()
+
+add_library(${PROJECT_NAME} INTERFACE)
+target_link_libraries(${PROJECT_NAME} INTERFACE
+  rclcpp::rclcpp
+  ${rmf_scheduler_msgs_LIBRARIES}
+)
+target_include_directories(${PROJECT_NAME} INTERFACE
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+  ${rmf_scheduler_msgs_INCLUDE_DIRS}
+)
+
+#===============================================================================
+
+if(BUILD_TESTING)
+  find_package(ament_cmake_catch2 REQUIRED)
+  find_package(ament_cmake_uncrustify REQUIRED)
+  find_package(rmf_utils REQUIRED)
+  find_package(Boost REQUIRED COMPONENTS filesystem)
```

Comment:
Just a nit, on docker builds this dependency is bringing in all of the boost library, I saw that in the other package std::filesystem is being used instead, any chance to also use it here?

Reply:
iirc, this is needed for the `boost::process`, it doesn't allow using paths from `std::filesystem`.